src/main/java/com/ericsson/ei/handlers/ExtractionHandler.java

```diff
@@ -78,23 +77,31 @@ public void runExtraction(RulesObject rulesObject, String mergeId, String event,
         try {
             JsonNode extractedContent = extractContent(rulesObject, event);
 
+            String mergedContent = null;
+            String aggregatedObjectId = null;
             if(aggregatedDbObject != null) {
                 LOGGER.debug("ExtractionHandler: Merging Aggregated Object:\n{}"
                         + "\nwith extracted content:\n{}"
                         + "\nfrom event:\n{}",
                         aggregatedDbObject.toString(), extractedContent.toString(), event);
-                String objectId = objectHandler.extractObjectId(aggregatedDbObject);
-                String mergedContent = mergeHandler.mergeObject(objectId, mergeId, rulesObject, event, extractedContent);
-                processRulesHandler.runProcessRules(event, rulesObject, mergedContent, objectId, mergeId);
+                aggregatedObjectId = objectHandler.extractObjectId(aggregatedDbObject);
+                mergedContent = mergeHandler.mergeObject(aggregatedObjectId, mergeId, rulesObject, event, extractedContent);
+                mergedContent = processRulesHandler.runProcessRules(event, rulesObject, mergedContent, aggregatedObjectId, mergeId);                
             } else {
+            	LOGGER.debug("***** Extraction starts for the Id: " + mergeId);
                 ObjectNode objectNode = (ObjectNode) extractedContent;
                 objectNode.put("TemplateName", rulesObject.getTemplateName());
-                mergeHandler.addNewObject(event, extractedContent, rulesObject);
+                mergedContent = mergeHandler.addNewObject(event, extractedContent, rulesObject);
+                aggregatedObjectId = mergeId;
                 upStreamEventsHandler.runHistoryExtractionRulesOnAllUpstreamEvents(mergeId);
+                mergedContent = objectHandler.findObjectById(mergeId);
+                LOGGER.debug("**** Extraction ends for the Id: " + mergeId);               
```

Comment:
Same comment for this log, which Id is it we're talking about here?

Reply:
should be trace log type.

Reply:
Updated